### PR TITLE
4 get api/reviews/reviewid

### PR DESCRIPTION
**Pull request by ahsan-0**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -77,3 +77,33 @@ describe("GET /api/reviews", () => {
...
 });
+
+describe("GET /api/reviews/:review_id", () => {
+  test("should respond with 200 and a review object with correct review id", () => {
+    return request(app)
+      .get("/api/reviews/2")
+      .expect(200)
+      .then(({ body }) => {
+        const { review } = body;
+        expect(body).toEqual({ review });
```
Comment (roisinoreilly) [inline __tests__/app.test.js]: body is always going to equal review as you just declared it in the line before, it's the same as saying a = b, does b equal a? You need to actually look into the body and make sure the outline of the object looks how you expect, use one of the jest matchers .toMatchObject or .objectContaining to make sure all the keys and values you expect are actually there

File: error-handling/erros.js [hunk 2]
```diff
@@ -1,4 +1,20 @@
-exports.handle500s = (err, req, res, next) => {
+exports.handle404s = (err, req, res, next) => {
   console.log(err);
-  res.status(500).send({ msg: "Internal Server Error" });
+  res.status(404).send({ msg: "Not Found" });
+};
+
+exports.idDoesNotExistError = (err, req, res, next) => {
```
Comment (roisinoreilly) [inline error-handling/erros.js]: a bit picky but I'm not the biggest fan of the name of this function, what about handleCustomErrors?

File: error-handling/erros.js [hunk 3]
```diff
@@ -1,4 +1,20 @@
...
+  if (err.status && err.msg) {
+    res.status(404).send({ msg: err.msg });
+  } else {
+    next(err);
+  }
+};
+
+exports.invalidIdError = (err, req, res, next) => {
```
Comment (roisinoreilly) [inline error-handling/erros.js]: this will handle other errors that PSQL sends back, how about handleSQLErrors to make it more generic?

**Review by roisinoreilly — CHANGES_REQUESTED** [review]
Great job, I'd just like you to refactor your test to check that the review you're receiving back looks how you expect, once you've done that send us another quick PR

File: __tests__/app.test.js [hunk 4]
```diff
@@ -77,3 +77,44 @@ describe("GET /api/reviews", () => {
...
+
+describe("GET /api/reviews/:review_id", () => {
+  test("should respond with 200 and a review object with correct review id", () => {
+    return request(app)
+      .get("/api/reviews/2")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body).toMatchObject({
+          review: expect.objectContaining({
+            review_id: expect.any(Number),
```
Comment (Poonam-raj) [inline __tests__/app.test.js]: You knwo exactly which review is coming back. Make sure you are specific on line 89

File: controllers/controllers.js [hunk 5]
```diff
@@ -10,6 +11,12 @@ exports.getCategories = (req, res, next) => {
     });
 };
 
+exports.getReviewById = (req, res, next) => {
+  const { review_id } = req.params;
+  selectReviewById(review_id)
+    .then((review) => {
+      res.status(200).send({ review });
+
```
Comment (Poonam-raj) [inline controllers/controllers.js]: You have nested getReviews within getReviewById, please unnest this

File: error-handling/erros.js [hunk 6]
```diff
@@ -1,4 +1,20 @@
...
+    next(err);
+  }
+};
+
+exports.handleSQLErrors = (err, req, res, next) => {
+  if (err.code === "22P02") {
+    res.status(400).send({ msg: "Invalid id input" });
+  } else {
+    next(err);
+  }
```
Comment (Poonam-raj) [inline error-handling/erros.js:19]: Please add a 500 error handler into your app - this should be the final error handler, where everything else ends up if it doesn't get caught by the other bits

**Review by Poonam-raj — APPROVED** [review]
Hi Ahsan, few amendments from my side, read through my comments and make the changes needed. 

No need for another PR though as the code is largely good, I trust you to change the small things and then merge yourself

**Pull request merged**